consensus/wal.go

```diff
@@ -83,7 +86,10 @@ func (wal *WAL) Save(wmsg WALMessage) {
 		}
 	}
 	// Write the wal message
-	var wmsgBytes = wire.JSONBytes(TimedWALMessage{time.Now(), wmsg})
+	innerMsgBytes := wire.JSONBytes(wmsg)
```

Comment:
I think we probably shouldn't use JSON here. It's kind of convenient for glancing at the wal, but for the sake of speed we should use a binary format. We can have a tool to map the binary lines to json for human readability.

Also, in that setting, I think we want the message size immediately after the checksum, ie. as Kyle put it "One simple approach to writing recoverable WALs is to write a fixed-size checksum, fixed-size length header, then variable-length record."

Reply:
By here you mean the whole WAL? or just this intermediate encoding?

Reply:
yeh both. certainly dont really see any utility to the outer encoding being json if the inner encoding is binary, so might as well do it all binary. Do you have a different opinion?

Reply:
No, it is just that we will lose readability. I guess if we will provide a wal2json-viewer script, which outputs WAL as json, it should be fine.

Reply:
doh, all `tmlibs/autofile/group` code is about strings (separated with "\n") including searching for "#ENDHEIGHT". I am rewriting it right now

Reply:
"You can open a Group to keep restrictions on an AutoFile, like
the maximum size of each chunk, and/or the total amount of bytes
stored in the group."

interesting. so now it is possible to have one part of an msg in file X and the rest of it in file Y.

Reply:
That doesn't sound good. Are you saying AutoFile won't work with binary format?

Reply:
No, I will work, but autofile/group won't. There is no `Write([]byte)` method to start with. And search functionality will be broken once we start using binary format.